pallets/pallet-bonded-coins/src/lib.rs

```diff
@@ -175,66 +182,193 @@ pub mod pallet {
 	impl<T: Config> Pallet<T> {
 		#[pallet::call_index(0)]
 		#[pallet::weight(Weight::from_parts(10_000, 0) + T::DbWeight::get().writes(1))]
-		pub fn create_pool(_origin: OriginFor<T>) -> DispatchResult {
-			todo!()
+		pub fn create_pool(
+			origin: OriginFor<T>,
+			curve: CurveInput<CurveParameterInputOf<T>>,
+			currencies: BoundedVec<TokenMetaOf<T>, T::MaxCurrencies>,
+			denomination: u8,
+			transferable: bool,
+		) -> DispatchResult {
+			let who = T::PoolCreateOrigin::ensure_origin(origin)?;
+
+			let currency_length = currencies.len();
+
+			let checked_curve = curve.try_into().map_err(|_| Error::<T>::InvalidInput)?;
+
+			let current_asset_id = NextAssetId::<T>::get();
+
+			let (currency_ids, next_asset_id) = Self::generate_sequential_asset_ids(current_asset_id, currency_length)?;
+
+			let pool_id = T::PoolId::from(currency_ids.blake2_256());
+
+			// Todo: change that.
+			T::DepositCurrency::hold(
+				&T::RuntimeHoldReason::from(HoldReason::Deposit),
+				&who,
+				T::BaseDeposit::get().saturating_add(
+					T::DepositPerCurrency::get()
+						.saturating_mul(currency_length.saturated_into())
+						.saturated_into(),
+				),
+			)?;
+
+			currencies
+				.iter()
+				.enumerate()
+				.try_for_each(|(idx, entry)| -> DispatchResult {
+					let asset_id: &T::AssetId = currency_ids.get(idx).ok_or(Error::<T>::IndexOutOfBounds)?;
+					T::Fungibles::create(asset_id.clone(), pool_id.clone().into(), false, entry.min_balance)?;
+
+					// set metadata for new asset class
+					T::Fungibles::set(
+						asset_id.clone(),
+						&pool_id.clone().into(),
+						entry.name.clone().into(),
+						entry.symbol.clone().into(),
+						denomination,
+					)?;
+
+					Ok(())
+				})?;
+
+			// Touch the pool account in order to be able to transfer the collateral currency to it
+			T::CollateralCurrency::touch(T::CollateralAssetId::get(), &pool_id.clone().into(), &who)?;
+
+			Pools::<T>::set(
+				&pool_id,
+				Some(PoolDetails::new(who, checked_curve, currency_ids, transferable)),
+			);
+
+			// update the storage for the next tx.
+			NextAssetId::<T>::set(next_asset_id);
+
+			Self::deposit_event(Event::PoolCreated { id: pool_id });
+
+			Ok(())
+		}
+
+		#[pallet::call_index(2)]
+		#[pallet::weight(Weight::from_parts(10_000, 0) + T::DbWeight::get().writes(1))]
+		pub fn reset_team(
+			origin: OriginFor<T>,
+			pool_id: T::PoolId,
+			team: PoolManagingTeam<AccountIdOf<T>>,
+			currency_idx: u32,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
```

Comment:
All the signed origins should now be migrated to the configurable `DefaultOrigin`
```suggestion
			let who = T::DefaultOrigin::ensure_origin(origin)?;
```

Reply:
introduced in: [30cbeb3](https://github.com/KILTprotocol/kilt-node/pull/768/commits/30cbeb34d5d6e3ae8a7561c38beef14206755a5d)